app/.eslintrc.js

```diff
@@ -0,0 +1,320 @@
+module.exports = {
+    "env": {
+        "es2020": true,
+        "node": true
+    },
+    "parser": "@typescript-eslint/parser",
+    "parserOptions": {
+        "project": "tsconfig.json"
+    },
+    "plugins": [
+        "@typescript-eslint"
+    ],
+    "extends": [
+        "eslint:recommended",
+        "plugin:@typescript-eslint/recommended",
+        "plugin:@typescript-eslint/recommended-requiring-type-checking"
+    ],
+    "rules": {
+        // @typescript-eslint rules
+        "@typescript-eslint/array-type": [
+            "error",
+            {
+                "default": "array-simple"
+            }
+        ],
+        "@typescript-eslint/consistent-type-definitions": [
```

Comment:
This needs further explanation

Reply:
This means preferring `interface` over `type` when declaring definitions for objects. Apart from a few exceptions we were already doing it like this, so it's good to keep it consistent.

Reply:
Well I see this as requiring the class to implement an interface, but the cases where it's come up, it makes sense to be a type?

Reply:
`type` is used for all kinds of type definitions, `interface` only for object definitions. Here is an article if you want more: https://pawelgrzybek.com/typescript-interface-vs-type/

Reply:
Upon reviewing the article, I've been swayed. Resolving.